WeScan/Scan/ScannerViewController.swift

```diff
@@ -137,46 +143,94 @@ final class ScannerViewController: UIViewController {
             activityIndicator.centerYAnchor.constraint(equalTo: view.centerYAnchor)
         ]
         
-        NSLayoutConstraint.activate(quadViewConstraints + cancelButtonConstraints + shutterButtonConstraints + activityIndicatorConstraints)
+        if #available(iOS 11.0, *) {
+            let window = UIApplication.shared.keyWindow
+            
+            toolbarConstraints = [
+                toolbar.widthAnchor.constraint(equalTo: view.widthAnchor),
+                toolbar.trailingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.trailingAnchor),
+                toolbar.leadingAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leadingAnchor),
+                toolbar.topAnchor.constraint(equalTo: view.topAnchor)
+            ]
+            
+            if let safeAreaInsets = window?.safeAreaInsets {
+                toolbarConstraints.append(toolbar.heightAnchor.constraint(equalToConstant: safeAreaInsets.top + 44.0))
+            }
+            
+            cancelButtonConstraints = [
+                cancelButton.leftAnchor.constraint(equalTo: view.safeAreaLayoutGuide.leftAnchor, constant: 24.0),
+                view.safeAreaLayoutGuide.bottomAnchor.constraint(equalTo: cancelButton.bottomAnchor, constant: (65.0 / 2) - 10.0)
+            ]
+            
+            let shutterButtonBottomConstraint = view.safeAreaLayoutGuide.bottomAnchor.constraint(equalTo: shutterButton.bottomAnchor, constant: 8.0)
+            shutterButtonConstraints.append(shutterButtonBottomConstraint)
+        } else {
+            toolbarConstraints = [
+                toolbar.widthAnchor.constraint(equalTo: view.widthAnchor),
+                toolbar.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+                toolbar.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+                toolbar.topAnchor.constraint(equalTo: view.topAnchor)
+            ]
+            
+            cancelButtonConstraints = [
+                cancelButton.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 24.0),
+                view.bottomAnchor.constraint(equalTo: cancelButton.bottomAnchor, constant: (65.0 / 2) - 10.0)
+            ]
+            
+            let shutterButtonBottomConstraint = view.bottomAnchor.constraint(equalTo: shutterButton.bottomAnchor, constant: 8.0)
+            shutterButtonConstraints.append(shutterButtonBottomConstraint)
+        }
+        
+        NSLayoutConstraint.activate(quadViewConstraints + cancelButtonConstraints + shutterButtonConstraints + activityIndicatorConstraints + toolbarConstraints)
     }
     
     // MARK: - Actions
     
+    @objc private func orientationChanged(_ notification: NSNotification) {
+        var rotation: CGFloat = 0
+        switch UIDevice.current.orientation {
+        case .portrait, .faceUp:
+            rotation = 0
+        case .landscapeLeft:
+            rotation = .pi / 2
+        case .landscapeRight:
+            rotation = -(.pi / 2)
```

Comment:
Nit: the parentesis are nit required here.

Reply:
I never remember about parentheses, but wouldn't that make it become the negative pi divided by 2?

Reply:
Yes you're correct, but for multiplications and divisions it doesn't matter: 
`-4/2 = 4/-2 = -(4/2) = -2` The convention is usually to put it on the numerator (at the top).

Reply:
Ahhhh yeah fair enough 🙃

Reply:
Removed this section as I removed the Cancel button transform